compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2842,6 +2842,14 @@ public boolean hasFillerValue(BType type) {
                 BTupleType tupleType = (BTupleType) type;
                 return tupleType.getTupleTypes().stream().allMatch(eleType -> hasFillerValue(eleType));
             default:
+                // filler value is 0
+                if (TypeTags.isIntegerTypeTag(type.tag)) {
+                    return true;
+                }
+                // TODO : clarify
```

Comment:
@KRVPerera, @rdhananjaya did we clarify/follow up on the concerns we had regarding filler values for XML?

Reply:
We agreed to handle `xml` type for now. Shall we create a followup issue after clarification?

Reply:
https://github.com/ballerina-platform/ballerina-spec/issues/490